### DISCUSSION TITLE
Unexpected Revert In compiling Test

### DISCUSSION THREAD

**Discussion by mihirWeb** [Cyfrin/foundry-full-course-cu#1331] [Q&A]
I write down simple to check whether the mapping from address to amount is correct but i am getting revert by EVM compiler and I don't know why, here is the code:-

In FundMeTest.t.sol file:-

```contract FumdMeTest is Test {
    FundMe fundMe;
    address USER = makeAddr("user");
    uint256 constant SEND_VALUE = 6 ether;
    function setUp() external {
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
        vm.deal(USER, 10 ether);
    }
    
    function testAddressToAmountFunded() public{
        vm.prank(USER);
        fundMe.fund{value: SEND_VALUE}();

        console.log(fundMe.MINIMUM_USD());
        console.log(SEND_VALUE);


        uint256 fundAmount = fundMe.getAddressToAmountFunded(USER);
        assertEq(fundAmount, SEND_VALUE);

        // this function is reverting and i dont know why
    }
    }```
    
    FundMe File:-
    
    ```contract FundMe {
    using PriceConverter for uint256;

    AggregatorV3Interface private s_priceFeed;
    mapping(address => uint256) private addressToAmountFunded;
    constructor(address priceFeed) {
        i_owner = msg.sender;
        s_priceFeed = AggregatorV3Interface(priceFeed);
    }

    function fund() public payable {
        require(msg.value.getConversionRate() >= MINIMUM_USD, "You need to spend more ETH!");
        addressToAmountFunded[msg.sender] += msg.value;
        funders.push(msg.sender);
    }
    function getAddressToAmountFunded(address funderAddress) external view returns(uint256){
        return addressToAmountFunded[funderAddress];
    } ```
    
    ERROR
    ```root@LAPTOP-FCMIQES2:/mnt/d/solidity/Foundry_demo/foundry-fund-me# forge test --match-test testAddressToAmountFunded -vvv
[⠒] Compiling...
[⠆] Compiling 1 files with 0.8.18
[⠰] Solc 0.8.18 finished in 2.11sCompiler run successful!
[⠔] Solc 0.8.18 finished in 2.11s

Running 1 test for test/FundMeTest.t.sol:FumdMeTest
[FAIL. Reason: EvmError: Revert] testAddressToAmountFunded() (gas: 19906)
Traces:
  [19906] FumdMeTest::testAddressToAmountFunded()
    ├─ [0] VM::prank(user: [0x6CA6d1e2D5347Bfab1d91e883F1915560e09129D])
    │   └─ ← ()
    ├─ [3070] FundMe::fund{value: 6000000000000000000}()
    │   ├─ [0] 0x694AA1769357215DE4FAC081bf1f309aDC325306::latestRoundData() [staticcall]
    │   │   └─ ← ()
    │   └─ ← EvmError: Revert
    └─ ← EvmError: Revert

Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 1.35ms```

**Comment by Loupsteb** [comment]
I've exactly the same problem than you, i hope we find a response.
If I find the answer i'll share with you

**Reply by mihirWeb** [reply]
I've figured out the problem, the problem is with the mockV3Aggregator, for some reason that mock aggregator is not working but if you do the same by forking url from sapolia or Ethereum mainnet then it will going to run without any problem

**Reply by Loupsteb** [reply]
I just find my error i forget to modify the DeployFundMe script file. So when i tried to deploy with Anvil the contract dont use the correct price feed, maybe verify your DeployFundMe.s.sol.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/52604984/f870643e-f9d7-48a1-92dc-083d293848a9)

**Reply by mihirWeb** [reply]
Mine is still showing that error, don't know why. By the way, why you are returning helperConfig along with fundMe? I tried to do the same by seeing yours but it is still showing an error, have a look
![Screenshot (346)](https://github.com/Cyfrin/foundry-full-course-f23/assets/106872436/e88cecd3-d734-49f6-b73f-8b4215f8073d)

**Reply by Loupsteb** [reply]
the problem is from the fund function.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/52604984/8eb46411-a5af-4315-b85a-206de6f9fa6a)
Add helperConfig, it's important because for deploy the deployFundMe you need to specify what network the contract need to use

**Reply by mihirWeb** [reply]
I just did it but my test is still reverting, can you share your FundMeTest file?

**Reply by Loupsteb** [reply]
ofc
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/52604984/dbb11c0d-3818-434a-b4cd-461be405b5be)

**Reply by mihirWeb** [reply]
I did but my tests are still reverting on Anvil
![Screenshot (348)](https://github.com/Cyfrin/foundry-full-course-f23/assets/106872436/e531a9aa-2eff-442a-aa52-6b6002014770)

but what is the need of pricefeed in the test....we only need it once while we are deploying the script and deploy script will call it itself?

**Reply by Loupsteb** [reply]
i thin the problem is your HelperConfig
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/52604984/6556a6e2-c867-475c-91b1-a9bb54e1670a)

**Reply by mihirWeb** [reply]
I checked the helperConfig code and it is same as yours but, it is still reverting🥲
